src/platform/Darwin/PlatformManagerImpl.cpp

```diff
@@ -147,7 +147,7 @@ CHIP_ERROR PlatformManagerImpl::_PostEvent(const ChipDeviceEvent * event)
 bool PlatformManagerImpl::_IsChipStackLockedByCurrentThread() const
 {
     // Assume our caller knows what they are doing in terms of concurrency if the work queue is suspended.
-    return IsWorkQueueCurrentQueue() || mWorkQueueState.load() == WorkQueueState::kSuspended;
+    return IsWorkQueueCurrentQueue() || mWorkQueueState.load() == WorkQueueState::kSuspended || (mChipStackIsLocked && (pthread_equal(pthread_self(), mChipStackLockOwnerThread)));
```

Comment:
are these equivalent?

The `Assume caller knows what they are doing in terms of concurrency if the work queue is suspended` already sounds like a reasonably strong assumption. Assuming that `mChipStackIsLocked` is mutually exclusive with the work queue seems unclear to me ... how do we know the work queue always locks the chip stack? At a first glance here we do not, we just implement separate locks and do not guarantee mutual exclusivity.